pandas/_libs/tslibs/conversion.pyx

```diff
@@ -524,10 +524,20 @@ cdef inline check_overflows(_TSObject obj):
     # GH#12677
     if obj.dts.year == 1677:
         if not (obj.value < 0):
-            raise OutOfBoundsDatetime
+            # GH#32967
+            from pandas._libs.tslibs.timestamps import Timestamp
```

Comment:
@jbrockmendel ok with this import (should it be just `from pandas import Timestamp`)? since inside an except not sure it matters much.

Reply:
slight preference for the more verbose import here